curator-framework/src/main/java/org/apache/curator/framework/state/ConnectionStateManager.java

```diff
@@ -340,4 +338,12 @@ private void setCurrentConnectionState(ConnectionState newConnectionState)
         currentConnectionState = newConnectionState;
         startOfSuspendedEpoch = (currentConnectionState == ConnectionState.SUSPENDED) ? System.currentTimeMillis() : 0;
     }
+
+    private synchronized int getUseSessionTimeoutMs() {
```

Comment:
Looks like the comment on `startOfSuspendedEpoch` "guarded by sync" is a lie. `getUseSessionTimeoutMs()` doesn't need to be synchronized if we make `startOfSuspendedEpoch` volatile.

Reply:
I think volatile is cleaner. I don't see any read/increments so it doesn't need to be AtomicLong either.

Reply:
Perfect.

Reply:
I've made it volatile now.